pkg/vm/vm.go

```diff
@@ -1055,6 +1055,11 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 				panic("zero modulus") // https://docs.microsoft.com/en-us/dotnet/api/system.numerics.biginteger.modpow?view=net-6.0#exceptions
 			}
 			res.Exp(base, exponent, modulus)
+
+			if base.Sign() < 0 && exponent.Bit(0) == 1 {
+				absModulus := new(big.Int).Abs(modulus)
+				res.Sub(res, absModulus)
```

Comment:
Please add a reference or a comment with reference or the number of issue, because otherwise it's not clear why this code exists.